### PR TITLE
Add function to validate DSTX message

### PR DESCRIPTION
**Pull request by rkarthik2k21**
This PR is part of addressing issue 4078 - Test DSTX validation during propogation #4078 (https://github.com/dashpay/dash/issues/4078) 

A function ValidateDSTX() has been added that essentially captures the existing code to validate a DSTX message. A unit test can be created to call this method in order for testing DSTX validation

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
still needs rebase

**Comment by rkarthik2k21** [comment]
I am not sure what went wrong. I created this branch from develop (and felt I had done a pull before that). I can create a new PR

**Comment by PastaPastaPasta** [comment]
You'll need to fetch latest develop. something like `git fetch upstream` assuming you've done `git remote add upstream https://github.com/dashpay/dash`

Then `git rebase upstream/develop` and `git push -f` should fix it

File: src/net_processing.cpp [hunk 1]
```diff
@@ -4908,5 +4924,9 @@ class CNetProcessingCleanup
         mapOrphanTransactionsByPrev.clear();
         nMapOrphanTransactionsSize = 0;
     }
+<<<<<<< HEAD
```
Comment (UdjinM6) [inline src/net_processing.cpp:4927]:
```suggestion
```

File: src/net_processing.cpp [hunk 2]
```diff
@@ -4908,5 +4924,9 @@ class CNetProcessingCleanup
...
 };
 static CNetProcessingCleanup instance_of_cnetprocessingcleanup;
+=======
+} instance_of_cnetprocessingcleanup;
+>>>>>>> Add function to validate DSTX message
```
Comment (UdjinM6) [inline src/net_processing.cpp:4932]:
```suggestion
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/net_processing.cpp (hunk 1 above)
Comment (rkarthik2k21) [reply src/net_processing.cpp]: Done

File: src/net_processing.cpp (hunk 2 above)
Comment (rkarthik2k21) [reply src/net_processing.cpp]: Done

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/net_processing.cpp [hunk 3]
```diff
@@ -3149,49 +3199,14 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
...
-
-            if (!dstx.CheckSignature(dmn->pdmnState->pubKeyOperator.Get())) {
-                LogPrint(BCLog::COINJOIN, "DSTX -- CheckSignature() failed for %s\n", hashTx.ToString());
-                return false;
-            }
 
-            LogPrint(BCLog::COINJOIN, "DSTX -- Got Masternode transaction %s\n", hashTx.ToString());
-            mempool.PrioritiseTransaction(hashTx, 0.1*COIN);
-            mmetaman.DisallowMixing(dmn->proTxHash);
+           // Validate DSTX and return bRet if we need to return from here
```
Comment (UdjinM6) [inline src/net_processing.cpp:3203]:
```suggestion
           // Validate DSTX and return bRet if we need to return from here
```

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**